proxy/coco/src/peer.rs

```diff
@@ -725,6 +748,30 @@ mod test {
         Ok(())
     }
 
+    /// Verify that asking the network for an unkown urn returns no providers.
+    #[tokio::test]
+    async fn get_urn_providers_none() -> Result<(), Error> {
+        let tmp_dir = tempfile::tempdir().expect("failed to create temdir");
+        let key = SecretKey::new();
+        let config = config::default(key, tmp_dir.path())?;
+        let api = Api::new(config).await?;
+        let unkown_urn = uri::RadUrn {
+            id: Hash::hash(b"project0"),
+            proto: uri::Protocol::Git,
+            path: "user/imperative-language"
+                .parse::<uri::Path>()
+                .map_err(Error::from)?,
+        };
+
+        let mut peers = api.providers(unkown_urn).await;
+        assert!(
+            peers.next().await.is_none(),
```

Comment:
I would be surprised if this ever returns. Perhaps the `upstream` wrapper should take a timeout or deadline parameter, and give up polling the stream once that elapses?

Reply:
What triggered that suspicion for you? Cause I had just realized that this assert always succeeds. I thought it might have been related to your point above on `await` but that change didn't make a difference here.

Reply:
I'm talking nonsense. You are right, it never returns, that's why I couldn't make it fail. Gotta fix https://github.com/radicle-dev/radicle-upstream/pull/875#discussion_r484291916.

Reply:
It's that the returned `Stream` will just be in pending state forever when there is nobody to answer. Of course it could also be the caller's responsibility to ensure it doesn't wait indefinitely, but it might also be the right place to _force_ them to think about this.